enhancements/machine-config/install-time-support-for-image-mode-on-openshift.md

```diff
@@ -0,0 +1,343 @@
+2. Cluster admin adds the `pull secret` needed to pull the container image to the cluster's `global pull-secret`
+3. Cluster admin runs `openshift-install create manifests` to generate the base Kubernetes manifests
+4. Cluster admin places `MachineOSConfig` (MOSC) YAML files in the `<install-dir>/manifests/` directory for any MachineConfigPools they want to use the Image Mode workflow for
+    - MOSC has annotation pointing to the pre-built container image (seeding the image to be used at install time)
+    - MOSC has the custom Containerfile used when building the custom OS container image in step 1
+    - MOSC has `pushSecret` set with the correct credentials needed to push any new images built post install to the registry
+5. Cluster admin runs `openshift-install create cluster` which automatically:
+   - Creates ignition configs by base64 encoding all YAML files from `manifests/` directory
+
+#### Step 2: Bootstrap Phase
+
+1. Bootstrap MCO recognizes MachineOSConfig manifests and stores them for later API creation
+2. Bootstrap MCO creates rendered MachineConfigs for each pool
+3. Bootstrap MCO then does the following to inject the pre-built container image into the rendered MC
+    - Maps MachineOSConfigs with pre-built annotations to their target pools
+    - Validates pre-built image format (digest format required)
+    - Sets `OSImageURL = preBuiltImage` in rendered MachineConfigs for pools with pre-built images and MachineOSConfigs
+    - Adds tracking annotation to modified MachineConfigs
+4. Bootstrap MCO creates ignition files with modified rendered MCs containing `OSImageURL` pointing to the pre-built container image
+5. Bootstrap MCO writes MachineOSConfig manifests to `/etc/mcs/bootstrap/machine-os-configs/` for post-bootstrap API creation
+6. The MCD first boot service detects the different `OSImageURL` value and uses `rpm-ostree rebase` to rebase the node to the custom pre-built container image
+7. After the cluster bootstrap completes, MCO starts and begins post-bootstrap processing
+
+#### Step 3: Post-Bootstrap Seeding Phase
+
+1. MCO controller's syncMachineOSConfigs() function reads MachineOSConfig files from `/etc/mcs/bootstrap/machine-os-configs/`
+2. MCO creates MachineOSConfig objects in the Kubernetes API if they don't already exist
+3. Build controller detects the `machineconfiguration.openshift.io/pre-built-image` annotation on the newly created MachineOSConfig
+4. Build controller triggers seeding workflow by calling `seedMachineOSConfigWithExistingImage()``
+5. MCO creates a synthetic MachineOSBuild with:
+    - Successful status referencing the seeded pre-built OS container image
+    - `PreBuiltImage` label set to "true"
+    - Proper metadata including MOSC name, DigestedImagePushSpec, and MCP reference
+    - Success condition with reason "PreBuiltImageSeeded"
+6. MCO updates MachineOSConfig status with `CurrentImagePullSpec` pointing to the pre-built image
```

Comment:
I feel like this section has a race condition with serving worker configs perhaps?

We'd also need to have the eliminate additional reboot functionality for that to work. And we'd have to make worker workflow the same, otherwise as I understand it the bootstrap->master will be populating osImageURL whereas the workers (and in-cluster master scaleups) will be using the desiredImage annotation, which will be weird to have that split.

(Alternatively maybe we should just expand on worker details here)